### PR TITLE
Images not imported [SCI-1807]

### PR DESCRIPTION
**Pull request by Zanz2**
Images are now displayed in: protocol description, before_start, warnings,guidelines, manuscript_citation, and also in Step description and step expected result

(the html  tag is whitelisted in these components)

### CODE REVIEW

**Comment by Zanz2** [comment]
I made the mistake in the original pull request of this, where before i pulled from master while developing, so it merged them again, and the pull request got huge, i closed that PR and made this one, i now know what not to do anymore.

File: app/helpers/protocols_io_helper.rb [hunk 1]
```diff
@@ -280,10 +301,13 @@ def protocols_io_fill_desc(json_hash)
         description_string += '<br>'
       elsif json_hash[e].present?
         new_e = '<strong>' + e.humanize + '</strong>'
+
+        image_tag = Array(nil)
+        image_tag = Array('img') if allowed_image_attributes.include? e
```
Comment (mlorb) [inline app/helpers/protocols_io_helper.rb]: Here the same, you can use ternary operator.

File: app/helpers/protocols_io_helper.rb [hunk 2]
```diff
@@ -146,12 +146,20 @@ def close_open_html_tags(text)
...
-  def prepare_for_view(attribute_text1, size, table = 'no_table')
+  # Images are allowed in:
+  # Step: description, expected result
+  # Protocol description : description before_start warning
+  # guidelines manuscript_citation
+  def prepare_for_view(
+    attribute_text1, size, table = 'no_table', image_allowed = false
+  )
+    image_tag = Array(nil)
+    image_tag = Array('img') if image_allowed
```
Comment (mlorb) [inline app/helpers/protocols_io_helper.rb]: Here the same, you can use ternary operator.

File: app/controllers/protocols_controller.rb [hunk 3]
```diff
@@ -601,8 +601,17 @@ def import
   def protocolsio_import_create
     @protocolsio_too_big = false
     @protocolsio_invalid_file = false
+    @protocolsio_no_file = false
```
Comment (mlorb) [inline app/controllers/protocols_controller.rb]: This shouldn't be part of this PR.

File: app/views/protocols/protocolsio_import_create.js.erb [hunk 4]
```diff
@@ -1,11 +1,16 @@
...
+    $('#pio_no_file_error_span').empty();
+})
 <% if @protocolsio_too_big %>
 $('#modal-import-json-protocol').modal('hide');
 HelperModule.flashAlertMsg(' <%= t('my_modules.protocols.load_from_file_size_error',
                  size: Constants::FILE_MAX_SIZE_MB ) %>','danger');
 <% elsif @protocolsio_invalid_file %>
 $('#modal-import-json-protocol').modal('hide');
 HelperModule.flashAlertMsg(' <%= t('my_modules.protocols.load_from_file_invalid_error') %>','danger');
+<% elsif @protocolsio_no_file %>
```
Comment (mlorb) [inline app/views/protocols/protocolsio_import_create.js.erb]: Same here. This shouldn't be part of this PR.
